File: gracie.lee/src/main/java/homework/chapter_5/problem2_1/Member.java
package homework.chapter_5.problem2_1;

import jakarta.persistence.*;

@Entity(name = "P21Member")
@Table(name = "P21_MEMBER")
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "MEMBER_ID")
    private Long id;

    private String name;

    @ManyToOne
    @JoinColumn(name = "TEAM_ID")
    private Team team;

    protected Member() {
    }

    public Member(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Team getTeam() {
        return team;
    }

    public void setTeam(Team team) {
        this.team = team;
    }

    // toString에서 team 출력 → 무한루프 발생!
    @Override
    public String toString() {
        return "Member{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", team=" + team +  // ← Team.toString() 호출 → 무한루프!
                '}';
    }
}

File: hyun.k/spring-boot/src/main/kotlin/homework/chapter_12/entity/Member.kt
package homework.chapter_12.entity

import jakarta.persistence.*

@Entity
class Member(
    @Column(nullable = false)
    var name: String,

    val age: Int,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    val team: Team? = null,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
)